cpp/src/arrow/compute/kernels/vector_selection.cc

```diff
@@ -2355,6 +2358,76 @@ const FunctionDoc array_take_doc(
      "given by `indices`.  Nulls in `indices` emit null in the output."),
     {"array", "indices"}, "TakeOptions");
 
+const FunctionDoc indices_nonzero_doc(
+    "Return indices of the array containing non zero or false values",
```

Comment:
I still find this summary line a bit confusing ("indices of the array" sounds a bit strange, and the "non zero or false" isn't necessarily read as "non-zero or non-false)" but rather as "non-zero or false) 

Maybe something like: `Return the indices of the values in the array that are non-zero` ?

Reply:
applied suggestion